e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -274,5 +276,59 @@ internal static class TestHelper {
             bool isSuccess = json.isSuccess;
             Assert.True(isSuccess);
         }
+
+        /// Wait for all API microservices of IIoT platform to be healthy.
+        /// </summary>
+        /// <param name="config"> IIoT platform configuration </param>
+        /// <param name="ct"> Cancellation token </param>
+        /// <returns></returns>
+        public static async Task WaitForServicesAsync(
+            IIIoTPlatformConfig config,
+            CancellationToken ct
+        ) {
+            const string healthyState = "Healthy";
+
+            var healthRoutes = new string[] {
+                TestConstants.APIRoutes.RegistryHealth,
+                TestConstants.APIRoutes.PublisherHealth,
+                TestConstants.APIRoutes.TwinHealth,
+                TestConstants.APIRoutes.HistoryHealth,
+                TestConstants.APIRoutes.GatewayHealth,
+                TestConstants.APIRoutes.VaultHealth,
+                TestConstants.APIRoutes.EventsHealth,
+                TestConstants.APIRoutes.JobOrchestratorHealth
+            };
+
+            var client = new RestClient(config.BaseUrl) {
+                Timeout = 60 * 1000
+            };
+
+            while(true) {
+                ct.ThrowIfCancellationRequested();
```

Comment:
Could we call this method only once outside the loop?

Reply:
It would not make sense to have it outside the loop as then it will be checked only once. What we need is to check for cancelation before initiating long-running operations.